compilers/concrete-optimizer/concrete-optimizer/src/optimization/dag/multi_parameters/analyze.rs

```diff
@@ -472,6 +526,37 @@ pub mod tests {
         }
     }
 
+    #[test]
+    #[should_panic(
+        expected = "You used the `allow_composition` flag when compiling this circuit, which allows to use the output of a circuit as its input without decryption. In the circuit you compiled, the output noise is not independant from the input noise. You can fix this by adding a refresh before returning from the circuit."
+    )]
+    fn test_composition_with_input_fails() {
+        let mut dag = unparametrized::OperationDag::new();
+        let _ = dag.add_input(1, Shape::number());
+        let p_cut = PrecisionCut { p_cut: vec![2] };
+        let _ = super::analyze(&dag, &CONFIG, &Some(p_cut), LOW_PRECISION_PARTITION, true);
+    }
+
+    #[test]
+    fn test_composition() {
```

Comment:
test_composition_1_partition
can you add a test with several partition where the output is used as input and added to another partition.

Reply:
Thanks for pointing that out, I was not sure about which test to add for exhaustiveness. I'll try to add proper circuits, but maybe I'll need a hand to craft all the needed ones.